#!/usr/bin/env python
PACKAGE = "delta_manipulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("v_max_fcu", double_t, 0, "maximum_drone_x,y,z_movement_speed",    0.5, 0.0,   1.0)
gen.add("yaw_max_fcu", double_t, 0, "maximum_drone_yaw_rate",    1.0, 0.0,   2.0)
gen.add("v_max_tooltip", double_t, 0, "maximum_tooltip_x,y,z_movement_speed",    0.1, 0.0,   1.0)

gen.add("v_traj_drone", double_t, 0, "trajectory_speed",    0.2, 0.0,   1.0)
gen.add("r_traj_drone", double_t, 0, "trajectory_radius",    0.8, 0.0,   1.0)
gen.add("h_traj_drone", double_t, 0, "trajectory_altitude",    1.0, 0.0,   3.0)

gen.add("v_traj_delta", double_t, 0, "trajectory_speed",    0.5, 0.0,   1.0)
gen.add("r_traj_delta", double_t, 0, "trajectory_radius",    0.3, 0.0,   1.0)

mode_enum = gen.enum([gen.const("static", int_t, 0, "static position hold"),
                 gen.const("line_z", int_t, 1, "line trajectory in z"),
                 gen.const("line_y", int_t, 2, "line trajectory in y"),
                 gen.const("line_x", int_t, 3, "line trajectory in x"),
                 gen.const("circle_xy", int_t, 4, "circular trajectory"),
                 gen.const("circle_yz", int_t, 5, "yz circle")],
                 "select trajectory type")

gen.add("delta_mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

mode_enum = gen.enum([gen.const("static", int_t, 0, "static position hold"),
                 gen.const("line_z", int_t, 1, "line trajectory in z"),
                 gen.const("line_y", int_t, 2, "line trajectory in y"),
                 gen.const("line_x", int_t, 3, "line trajectory in x"),
                 gen.const("circle_xy", int_t, 4, "circular trajectory"),
                 gen.const("figure_8", int_t, 5, "figure_8 trajectory")],
                 "select trajectory type")

gen.add("drone_mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

gen.add("x_max_pos", double_t, 0, "positive_x_boundary",    2.0, 0.0,   10.0)
gen.add("x_max_neg", double_t, 0, "negative_x_boundary",    2.0, 0.0,   10.0)
gen.add("y_max_pos", double_t, 0, "positive_y_boundary",    2.0, 0.0,   10.0)
gen.add("y_max_neg", double_t, 0, "negative_y_boundary",    2.0, 0.0,   10.0)
gen.add("z_max", double_t, 0, "z_boundary",    2.0, 0.0,   10.0)

exit(gen.generate(PACKAGE, "delta_manipulator", "Joystick"))